materials/for_loops_R.md

```diff
@@ -63,16 +79,29 @@ biomass_data <- data.frame(exper1 = c(24, 32, 62),
                            exper3 = c(1, 5, 3))
 
 for (exp_biomass in biomass_data) {
-  npp <- sum(19.3 * exp_biomass ** 2)
-  print(npp)
+  npp <- 19.3 * exp_biomass ** 2
+  total_npp <- sum(npp)
+  print(total_npp)
 }
 ```
 
-***Complete through exercise 4***
+> Do [Exercise 4 - for Loop]({{ site.baseurl }}/exercises/Functions-for-loop-R).
 
-## Apply
+> Make sure students get the basics before moving on.
 
-* Short cuts for simple loops
+### Alternate loops
+
+* [`apply()`](http://finzi.psych.upenn.edu/R/library/base/html/apply.html)
+    * Short cut for simple loops over rows and columns
+    * Other versions allow you various control options
+        * [`lapply()`](http://finzi.psych.upenn.edu/R/library/base/html/lapply.html): Operate across lists and vectors
+        * [`mapply()`](http://finzi.psych.upenn.edu/R/library/base/html/mapply.html): Pass multiple variables or function arguments
+        * [`rapply()`](http://finzi.psych.upenn.edu/R/library/base/html/rapply.html): Recursive operation
```

Comment:
I think getting into recursion is out of scope. Let's add `sapply` here instead since that's often what folks really want.